docs/changelog/104683.yaml

```diff
@@ -0,0 +1,5 @@
+pr: 104683
+summary: "Feature: re-arrange document id bytes to take advantage of shared prefixes"
```

Comment:
Perhaps rephrase this to emphasize that this reduces storage usage for the _id field? A trade off that works well for logging use cases?